text/000-rustdoc-cfgs-handling.md

```diff
@@ -0,0 +1,283 @@
+Rustdoc: stabilization of the `doc(cfg*)` attributes
+
+- Features Name: `doc_cfg`
+- Start Date: 2022-12-07
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#43781](https://github.com/rust-lang/rust/issues/43781)
+
+
+# Summary
+[summary]: #summary
+
+This RFC aims at providing rustdoc users the possibility to add visual markers to the rendered documentation to know under which conditions an item is available (currently possible through the following unstable features: `doc_cfg`, `doc_auto_cfg` and `doc_cfg_hide`).
+
+It does not aim to allow having a same item with different `cfg`s to appear more than once in the generated documentation.
+
+It does not aim to document items which are *inactive* under the current configuration (i.e., “`cfg`'ed out”).
+
+# Motivation
+[motivation]: #motivation
+
+The goal of this RFC is to stabilize the possibility to add visual markers to the rendered documentation to know under which conditions an item is available.
+
+Providing this information to users will solve a common issue: “Why can I see this item in the documentation and yet can't use it in my code?”.
+The end goal being to provide this information automatically so that the documentation maintenance cost won't increase.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC proposes to add the following attributes:
+
+  * `#[doc(cfg(...))]`
+
+    This attribute is used to document the operating systems, feature flags, and build profiles where an item is available. For example, `#[doc(cfg(unix))` will add a tag that says "this is supported on **unix** only" to the item.
+
+    The syntax of this attribute is the same as the syntax of the [`#[cfg(unix)]` attribute][cfg attribute] used for conditional compilation.
+
+  * `#[doc(auto_cfg)]`/`#[doc(no_auto_cfg)]`
+
+    When this is turned on, `#[cfg]` attributes are shown in documentation just like `#[doc(cfg)]` attributes are.
+
+  * `#[doc(cfg_hide(...))]` / `#[doc(cfg_show(...))]`
+
+    These attributes suppress or un-suppress the `auto_cfg` behavior for a particular configuration predicate.
+
+    For example, `#[doc(cfg_hide(windows))]` shall be used in newer versions of the [`windows` crate] to prevent the "this is supported on **windows** only" tag from being shown on every single item.
```

Comment:
```suggestion
    For example, `#[doc(cfg_hide(windows))]` could be used in newer versions of the [`windows` crate] to prevent the "this is supported on **windows** only" tag from being shown on every single item.
```

"Shall" would imply that they're being forced to do so, which isn't the case.